### DISCUSSION TITLE
[New Feature] Building Build Queue

### DISCUSSION THREAD

**Discussion by Onidotmoe** [Vinifera-Developers/Vinifera#1368] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

OpenRA has this neat feature where you can queue up buildings, tho in OpenRA the next one in the queue will not start until the one that is complete has been placed, this is nice but having a option for the queue to just continue would be interesting.

### Additional Context

![image](https://user-images.githubusercontent.com/32226493/183285164-84adbfca-d68b-430c-9517-34125bf39a4d.png)

**Comment by AtifArshad1995** [comment]
While I don't know about OpenRA, I do know that in YR-Phobos, there is this added in rulesmd.ini

[General]
BuildingProductionQueue=false

In Phobos, the **BuildingProductionQueue** is added and by default set to false. When it is set to true, we can queue buildings as much as defined by **MaximumQueuedObjects** in rulesmd

However, I do think it would be better to have a separate queue limit for buildings as it would provide a finer control. So, maybe have it like this would be better:

[General]
MaximumQueuedBuildings=XX           ; default is 0. Can be set from 0 to 99 or whatever
